Add Protobuf 'any' type support to Kafka connector (#17394)
trinodb/trino: docs/src/main/sphinx/connector/kafka.rst

``ARRAY``                             Protobuf type with ``repeated`` field
``MAP``                               ``Map``
``TIMESTAMP``                         ``Timestamp``, predefined in ``timestamp.proto``
``JSON``                              ``oneof`` (Confluent table supplier only), ``Any``
===================================== =======================================

any
+++++

Protobuf schemas containing `google.protobuf.Any <https://protobuf.dev/programming-guides/proto3/#any>`_ ``
fields are mapped to a ``JSON`` field in Trino when
``kafka.protobuf-any-support-enabled`` is set to ``true``. Any message types
contain an arbitrary serialized message as bytes and a type URL to resolve
that message's type. The connector reads the contents of the URL in order to
determine the type descriptor for the message and convert the message to JSON.

For example, given the following Protobuf schema:

.. code-block:: text

    syntax = "proto3";

    import "google/protobuf/any.proto";

    message schema {
        google.protobuf.Any any_message = 1;
    }

> The corresponding Trino row is a ``JSON`` field ``any_message``

Reviewer: cool

``ARRAY``                             Protobuf type with ``repeated`` field
``MAP``                               ``Map``
``TIMESTAMP``                         ``Timestamp``, predefined in ``timestamp.proto``
``JSON``                              ``oneof`` (Confluent table supplier only), ``Any``
===================================== =======================================

any
+++

Message types with an `Any <https://protobuf.dev/programming-guides/proto3/#any>`_
field contain an arbitrary serialized message as bytes and a type URL to resolve

> that message's type, typically with a scheme of ``file://`` or ``http://``.

Author: No, I've changed the docs to be more clear.

The connector reads the contents of the URL to create the type descriptor
for the ``Any`` message and convert the message to JSON. This behavior is enabled
by setting ``kafka.protobuf-any-support-enabled`` to ``true``.

The descriptors for each distinct URL are cached for performance reasons and
any modifications made to the type returned by the URL requires a restart of
Trino.

For example, given the following Protobuf schema which defines ``MyMessage``
with three columns:

.. code-block:: text

    syntax = "proto3";

    message MyMessage {
      string stringColumn = 1;
      uint32 integerColumn = 2;
      uint64 longColumn = 3;
    }

And a separate schema which uses an ``Any`` type which is a packed message
of the above type and a valid URL:

.. code-block:: text

    syntax = "proto3";

    import "google/protobuf/any.proto";

    message schema {
        google.protobuf.Any any_message = 1;
    }

> The corresponding Trino column will be named ``any_message`` of type ``JSON``

Reviewer: ```suggestion
The corresponding Trino column is named ``any_message`` of type ``JSON``
```
